internal/pkg/term/progress/cloudformation.go

```diff
@@ -85,6 +124,90 @@ func (c *regularResourceComponent) Done() <-chan struct{} {
 	return c.done
 }
 
+// ecsServiceResourceComponent can display an ECS service created with CloudFormation.
+type ecsServiceResourceComponent struct {
+	// Required inputs.
+	cfnStream    <-chan stream.StackEvent   // Subscribed stream to initialize the deploymentRenderer.
+	ecsDescriber stream.ECSServiceDescriber // Client needed to create an ECSDeploymentStreamer.
+	logicalID    string                     // LogicalID for the service.
+
+	// Optional inputs.
+	group      *errgroup.Group // Existing group to catch ECSDeploymentStreamer errors.
+	ctx        context.Context // Context for the ECSDeploymentStreamer.
+	renderOpts RenderOptions
+
+	// Sub-components.
+	resourceRenderer   Renderer
+	deploymentRenderer Renderer
+
+	done chan struct{}
+	mu   sync.Mutex
+}
+
+// Listen creates a deploymentRenderer if the service is being created, or updated.
+func (c *ecsServiceResourceComponent) Listen() {
+	var prevStatus string
+	for ev := range c.cfnStream {
+		if c.logicalID != ev.LogicalResourceID {
+			continue
+		}
+		curStatus := ev.ResourceStatus
+		if curStatus != prevStatus && cloudformation.StackStatus(curStatus).UpsertInProgress() {
+			c.mu.Lock()
+
+			// Start a deployment renderer if a service deployment is happening.
+			arn := ecs.ServiceArn(ev.PhysicalResourceID)
+			cluster, _ := arn.ClusterName() // Errors can't happen on valid ARNs.
+			service, _ := arn.ServiceName()
+			streamer := stream.NewECSDeploymentStreamer(c.ecsDescriber, cluster, service, ev.Timestamp)
+			c.deploymentRenderer = ListeningRollingUpdateRenderer(streamer, NestedRenderOptions(c.renderOpts))
+			c.group.Go(func() error {
+				return stream.Stream(c.ctx, streamer)
+			})
+
+			c.mu.Unlock()
+		}
+		prevStatus = curStatus
+	}
+	close(c.done)
+}
+
+// Render writes the status of the CloudFormation ECS service resource, followed with details around the
+// service deployment if a deployment is happening.
+func (c *ecsServiceResourceComponent) Render(out io.Writer) (numLines int, err error) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+
+	buf := new(bytes.Buffer)
+
+	nl, err := c.resourceRenderer.Render(buf)
+	if err != nil {
+		return 0, err
+	}
+	numLines += nl
+
+	var deploymentRenderer Renderer = &noopComponent{}
+	if c.deploymentRenderer != nil {
+		deploymentRenderer = c.deploymentRenderer
+	}
+
+	nl, err = deploymentRenderer.Render(buf)
+	if err != nil {
+		return 0, err
+	}
+	numLines += nl
+
+	if _, err = buf.WriteTo(out); err != nil {
```

Comment:
Writing an empty buffer doesn't change the output at all?

Reply:
Nope, nothing happens